Octokit.Tests/Clients/RepositoryContentsClientTests.cs

```diff
@@ -68,35 +68,394 @@ public class TheGetReadmeHtmlMethod
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, null, null, null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, null, null, ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, null, "readme.md", null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, null, "readme.md", ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, "readme.md", null, null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, "readme.md", null, ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, "readme.md", "master", null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", null, "readme.md", "master", ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", null, null, null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", null, null, ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", null, "master", null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", null, "master", ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", "readme.md", null, null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", "readme.md", null, ApiOptions.None));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContentsByRef("owner", "name", "readme.md", "master", null));
+                
+                
+                // empty string checks
+
+                // public Task<IReadOnlyList<RepositoryContent>> GetAllContents(string owner, string name)
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllContents("", ""));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllContents("", "name"));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllContents("owner", ""));
+
+                // public Task<IReadOnlyList<RepositoryContent>> GetAllContents(string owner, string name, ApiOptions options)
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllContents("", "", (ApiOptions)null));
```

Comment:
Same on all the empty string checks as above comment on null checks.  Each parameter needs to be checked, but not combinations of multiple parameters